egg/platform_x86.py

```diff
@@ -836,6 +972,69 @@ def lnot1(simd_ext, typ):
             return 'return (__mmask{le})(~{in0});'.format(**fmtspec)
     return not1(simd_ext, typ, True)
 
+# -----------------------------------------------------------------------------
+## Code for constant true
+
+def true0(simd_ext, typ, logical=False):
+    if typ == 'f16':
+        return \
+        '''nsimd_{simd_ext}_v{logi}f16 ret;
+           nsimd_{simd_ext}_vf32 cte =
+             {pre}castsi{nbits}_ps({pre}set1_epi8(-1));
+           ret.v0 = cte;
+           ret.v1 = cte;
+           return ret;'''.format(logi='l' if logical else '', **fmtspec)
+    elif typ in ['f32', 'f64']:
+        return '''return {pre}castsi{nbits}{suf}(
+                           {pre}set1_epi8(-1));'''.format(**fmtspec)
+    else:
+        return '''return {pre}set1_epi8(-1);'''.format(**fmtspec)
```

Comment:
I do not know which one is the fastest :
- the `*_set1_*` intrinsics or
- the `cmpeq(setzero(), setzero()) approach
as the second one does read from memory.

Reply:
My assumption was that the memory load could be averted if the value is kept in the register (if it is used multiple times). However, that ties up a register, whereas `setzero()` doesn't. Also, GCC converts `set1(-1)` for integer types to a compare. I'll switch to using a comparison instead.